django/db/backends/mysql/compiler.py

```diff
@@ -36,7 +36,21 @@ def as_sql(self):
 
 
 class SQLUpdateCompiler(compiler.SQLUpdateCompiler, SQLCompiler):
-    pass
+    def as_sql(self):
+        query = super().as_sql()
+        order_by_annotations = self.query.order_by
+        if not order_by_annotations:
+            return query
+
+        order_by_clause = 'ORDER BY '
+        for annotation in order_by_annotations:
+            order_by_clause += '`%s` %s' % (
+                (annotation[1:], 'DESC') if annotation[0] == '-' else (annotation, 'ASC')
+            )
+            if annotation != order_by_annotations[-1]:
+                order_by_clause += ', '
+        query_with_order_by_clause = ('%s  %s' % (query[0], order_by_clause))
+        return query_with_order_by_clause, query[1]
```

Comment:
You should use tuple deconstruction in a number of places, starting with `query, params = super().as_sql()`. See other SQLCompiler methods.

Also you seem to have replicated a bit of logic from `SQLCompiler.get_order_by()`. You shouldn't do that, but find a way to reuse it. `order_by()` supports many forms beyond the asc/desc field name form you've compiled here.

Reply:
> order_by() supports many forms beyond the asc/desc field name form you've compiled here.

For example, you could add a test for `order_by(F('number') * -1))`.

Reply:
@adamchainz Thanks for pointing that out and I've applied these suggestions here: https://github.com/django/django/pull/13088/commits/b9f3ff953541ca2d79ad58ca4b8c4033e2006d2b

I didn't realize how much I was simplifying this, so please let me know if I should add more tests on top of the one @charettes suggested.